fvm/environment/programs_test.go

```diff
@@ -122,58 +138,60 @@ func Test_Programs(t *testing.T) {
 
 	var contractASnapshot *state.ExecutionSnapshot
 	var contractBSnapshot *state.ExecutionSnapshot
-	var txAView *delta.View = nil
+	var txASnapshot *state.ExecutionSnapshot
 
 	t.Run("contracts can be updated", func(t *testing.T) {
-		retrievedContractA, err := accounts.GetContract("A", addressA)
+		retrievedContractA, err := getTestContract(
+			mainSnapshot,
+			contractALocation)
 		require.NoError(t, err)
 		require.Empty(t, retrievedContractA)
 
 		// deploy contract A0
-		procContractA0 := fvm.Transaction(
-			contractDeployTx("A", contractA0Code, addressA),
-			derivedBlockData.NextTxIndexForTestingOnly())
-		err = vm.Run(context, procContractA0, mainView)
+		executionSnapshot, output, err := vm.RunV2(
+			context,
+			fvm.Transaction(
+				contractDeployTx("A", contractA0Code, addressA),
+				derivedBlockData.NextTxIndexForTestingOnly()),
+			mainSnapshot)
 		require.NoError(t, err)
 
-		retrievedContractA, err = accounts.GetContract("A", addressA)
+		mainSnapshot = mainSnapshot.Append(executionSnapshot)
+
+		retrievedContractA, err = getTestContract(
+			mainSnapshot,
+			contractALocation)
 		require.NoError(t, err)
 
 		require.Equal(t, contractA0Code, string(retrievedContractA))
 
 		// deploy contract A
-		procContractA := fvm.Transaction(
-			updateContractTx("A", contractACode, addressA),
-			derivedBlockData.NextTxIndexForTestingOnly())
-		err = vm.Run(context, procContractA, mainView)
+		executionSnapshot, output, err = vm.RunV2(
+			context,
+			fvm.Transaction(
+				updateContractTx("A", contractACode, addressA),
+				derivedBlockData.NextTxIndexForTestingOnly()),
+			mainSnapshot)
 		require.NoError(t, err)
-		require.NoError(t, procContractA.Err)
+		require.NoError(t, output.Err)
+
+		mainSnapshot = mainSnapshot.Append(executionSnapshot)
 
-		retrievedContractA, err = accounts.GetContract("A", addressA)
+		retrievedContractA, err = getTestContract(
+			mainSnapshot,
+			contractALocation)
 		require.NoError(t, err)
 
 		require.Equal(t, contractACode, string(retrievedContractA))
 
 	})
-
 	t.Run("register touches are captured for simple contract A", func(t *testing.T) {
-
-		// deploy contract A
```

Comment:
this was a bug in the test.  we've already deploy contract A (immediately above), and didn't check for output.Error here.